docs/ref/models/expressions.txt

```diff
@@ -445,15 +445,9 @@ SQL that is generated. Here's a brief example::
     class Count(Aggregate):
         # supports COUNT(distinct field)
         function = 'COUNT'
-        template = '%(function)s(%(distinct)s%(expressions)s)'
-
-        def __init__(self, expression, distinct=False, **extra):
-            super().__init__(
-                expression,
-                distinct='DISTINCT ' if distinct else '',
-                output_field=IntegerField(),
-                **extra
-            )
+        # template = '...' # write it if you need to change the template
+        output_field = IntegerField()
+        allow_distinct = True
```

Comment:
Current custom example (`COUNT` with distinct) is already builtin so we should probably provide different feasible example, maybe `MEDIAN`.

Reply:
`MEDIAN` is not SQL aggregate function [https://www.postgresql.org/docs/current/functions-aggregate.html](url) and [https://dev.mysql.com/doc/refman/8.0/en/group-by-functions.html](url).
I can use bit_and, but it will be worse in my opinion. What do you think?

Reply:
It depends on database, e.g. `MEDIAN` exists on Oracle, `BIT_AND` doesn't. Maybe something with "standard" function, e.g. `SumAbs`: :thinking: 
```python
class SumAbs(Aggregate):
    function = 'SUM'
    template = '%(function)s(%(distinct)sABS(%(expressions)s))'
    allow_distinct = False

Reply:
I am updating the example for the custom aggregate functions. Therefore, when using this function, the clause `GROUP BY` will be added. Example with the standart function would be incorrect. I believe that `Count` is best choice among existings ones. The fact, that `COUNT` is already buildin is not too bad.

Reply:
To be honest I don't understand why `SumAbs` is incorrect, `SUM(ABS(column))` should work fine.

IMO example with `COUNT` doesn't fit to its description:
> _"... but you can also **completely customize the SQL** that is generated. Here's a brief example"_

Maybe we can provide really similar example but with `ALL` instead of `DISTINCT`, e.g.
```python
    class Sum(Aggregate):
        # supports SUM(ALL field)
        function = 'SUM'
        template = '%(function)s(%(all_values)s%(expressions)s)'
        allow_distinct = False

        def __init__(self, expression, all_values=False, **extra):
            super().__init__(
                expression,
                all_values='ALL ' if all_values else '',
                **extra
            )
```
I know that aggregates with and without `ALL` work in exactly the same way but it is not important. The most important thing is that we provide an example that shows how users can customize SQL.

Reply:
@VVyacheslav What do you think? :point_up: